priam/src/main/java/com/netflix/priam/identity/InstanceIdentity.java

```diff
@@ -109,154 +117,162 @@ public InstanceInfo getInstanceInfo() {
-                                    ins.getInstanceId(),
-                                    ins.getId());
-                            if (ins.getInstanceId().equals(myInstanceInfo.getInstanceId())) {
-                                logger.info(
-                                        "[Alive]  found that this node is alive."
-                                                + " application: {}"
-                                                + ", id: {}"
-                                                + ", instance: {}"
-                                                + ", region: {}"
-                                                + ", host ip: {}"
-                                                + ", host name: {}"
-                                                + ", token: {}",
-                                        ins.getApp(),
-                                        ins.getId(),
-                                        ins.getInstanceId(),
-                                        ins.getDC(),
-                                        ins.getHostIP(),
-                                        ins.getHostName(),
-                                        ins.getToken());
-                                return ins;
-                            }
-                        }
-                        return null;
-                    }
-                }.call();
-
-        // Grab a dead token
-        if (null == myInstance) {
-            myInstance =
-                    new RetryableCallable<PriamInstance>() {
-
-                        @Override
-                        public PriamInstance retriableCall() throws Exception {
-                            PriamInstance result;
-                            result = deadTokenRetriever.get();
-                            if (result != null) {
-
-                                isReplace =
-                                        true; // indicate that we are acquiring a dead instance's
-                                // token
-
-                                if (deadTokenRetriever.getReplaceIp()
-                                        != null) { // The IP address of the dead instance to which
-                                    // we will acquire its token
-                                    replacedIp = deadTokenRetriever.getReplaceIp();
-                                }
-                            }
-
-                            return result;
-                        }
+        // Grab the token which was already assigned.
+        myInstance = grabAlreadyAssignedToken();
 
-                        @Override
-                        public void forEachExecution() {
-                            populateRacMap();
-                            deadTokenRetriever.setLocMap(locMap);
-                        }
-                    }.call();
+        // Grab a dead token.
+        if (myInstance == null) {
+            myInstance = grabDeadToken();
```

Comment:
Can you add INFO log statement when we move from one case to other? It is easier to debug in case of any failure.

Reply:
Yes, make sense. Done.